src/requests.rs

```diff
@@ -0,0 +1,229 @@
+use std::{thread, time::Duration};
+
+use log::{debug, error, info};
+use rand::rngs::ThreadRng;
+use reqwest::{blocking::Client, header::CONTENT_TYPE};
+
+use crate::{
+    data::{
+        random_gen_dust_concentration, random_gen_humidity, random_gen_pressure,
+        random_gen_temperature, AirPurity, Reading,
+    },
+    error::PostSchedulerError,
+    URL,
+};
+
+const DEFAULT_REQUEST_AMOUNT: u32 = 1;
+const DEFAULT_TIME_PER_REQUEST: Duration = Duration::from_secs(10);
+const DEFAULT_NUM_THREADS: u32 = 1;
+pub const MAX_NUM_THREADS: u32 = 10;
+
+#[derive(Clone, Copy)]
+pub struct PostSchedulerBuilder {
+    request_amount: Option<u32>,
```

Comment:
@AntoniosBarotsis I think we should also change nearly ever mention of a `request` to be `post` or something contextually relevant. Does it even make sense to have a `requests.rs` now? Consider changing it to `sensor_posts.rs` as on idea.

Reply:
I've left a few mentions of `request` where I feel like it makes sense, such as [here](https://github.com/Jim-Hodapp-Coaching/ambi_mock_client/blob/5782646641471f87fc0b032e2d9c2b8ddb1e89aa/src/sensor_posts.rs#L181-L186). Let me know what you think.

Reply:
@jhodapp, probably missed my comment here :)